### DISCUSSION TITLE
How to exclude an object from an array into another page?

### DISCUSSION THREAD

**Discussion by liebeckm** [neuefische/questions#557] [Web] [answered]
Dear all,

I am working on my app right now. On the first page is a list with an overview of all offered rides. I want to build a second page now, which appears if you choose one of the listed rides.

I tried to build a Component (CardForDetailPage) and a new page (detail-ride/index) for this and implemented the Component on the new page.
Unfortunately, the Card does not appear. 

1. Can someone explain to me why this is not working and maybe has a solution on this?
2. I am generally unsure if this is the right way. If I continue like this I would have to build a details page for every ride that is offered. I think there is maybe a smarter solution so I only have so build one page? If you have any ideas feel free to share! :)

[Here](https://github.com/liebeckm/capstone-project/tree/detail-page) you can find the code.
[Here ](https://github.com/liebeckm/capstone-project/issues/2) you can find the User Story.

**Comment by toshydev** [comment, marked as answer]
Hey, I'm not in your course so I'm not sure if it's ok for me to suggest an answer but I'll try anyway 😊

You're not linking to the specific detail pages in the `List` component. 
I can see an `<a>` in the main index but I'm not sure if that would work.  You could try using the `next/router` and get the `id` from the `router.query` and use this `id` to tell the details page which ride to show. 
for that you could use a `[id].js` file.

I hope this makes sense. First post in the discussion forum 🫣😁

Nice app idea btw💪

**Reply by liebeckm** [reply]
Thank you! :)
